### DISCUSSION TITLE
Add option to skip comment when no changes and no warnings

### DISCUSSION THREAD

**Discussion by takayamaki** [suzuki-shunsuke/tfcmt#773] [Deprecated (Feature Request)] [answered]
### Feature Overview

When this option is specified, tfcmt adds labels but does not create comments for unchanged targets.

### Why is the feature needed?

We have many tfstates in our repository.
They are independent but constitute a single product.

![image](https://user-images.githubusercontent.com/24884114/236999116-97e0c57f-409d-48f4-b343-041de4cd3fa5.png)

Current version of tfcmt creates comments even when there are no changes.
Therefore, I spend a lot of time looking for comments that have changes.

### Does the feature include Breaking Changes?

I do not think  so.

### Example Code

```console
$ tfcmt -var target:${{ matrix.environments.dir }} plan -patch -skip-no-changes -- terraform plan -no-color -lock=false
```

**Comment by suzuki-shunsuke** [comment]
Thank you for your proposal.
It makes sense.

I'm working on it.
- https://github.com/suzuki-shunsuke/tfcmt/pull/774

**Reply by suzuki-shunsuke** [reply]
Published a prerelease version. https://github.com/suzuki-shunsuke/tfcmt/releases/tag/v4.4.0-1

**Reply by suzuki-shunsuke** [reply]
@takayamaki Could you try [v4.4.0-1](https://github.com/suzuki-shunsuke/tfcmt/releases/tag/v4.4.0-1)?
If there is no problem, I'll merge #774 and release v4.4.0.

**Reply by takayamaki** [reply]
oh, I will try it today.
Take a minutes.

**Reply by takayamaki** [reply]
Almost good.

1. tfcmt creates a comment because there are "changes" in the result.
2. I fix the "changes" and push a new commit.
3. tfcmt updates the comment to "no changes" as expected even if no changes.

This behaver is my expected, thank you!

However, I seems better that to create comment when result has warnings even if no changes.
https://github.com/suzuki-shunsuke/tfcmt/blob/638294e99f69035cae7ded3feb5d28d7337c512d/pkg/notifier/github/plan.go#L108

How do  you seem about this?

**Reply by suzuki-shunsuke** [reply]
It makes sense.
I fixed and published [v4.4.0-2](https://github.com/suzuki-shunsuke/tfcmt/releases/tag/v4.4.0-2).

https://github.com/suzuki-shunsuke/tfcmt/commit/5fc2e78e4db236a7f97ec7a4198bebe072b8fc85

**Reply by takayamaki** [reply]
Perfect, thank you!
Looks good to me.
![image](https://github.com/suzuki-shunsuke/tfcmt/assets/24884114/dbc8b823-87f4-48d9-a235-9bf1d59af490)

**Reply by suzuki-shunsuke** [reply]
Thank you.
I'll release v4.4.0.

docs: https://github.com/suzuki-shunsuke/tfcmt-docs/pull/57

**Reply by suzuki-shunsuke** [reply, marked as answer]
Released. https://github.com/suzuki-shunsuke/tfcmt/releases/tag/v4.4.0
https://suzuki-shunsuke.github.io/tfcmt/skip-no-changes
tfcmt has supported this feature.
